packages/ra-core/src/routing/useGetPathForRecord.ts

```diff
@@ -51,23 +51,7 @@ export const useGetPathForRecord = <RecordType extends RaRecord = RaRecord>(
     }
     const getPathForRecord = useGetPathForRecordCallback<RecordType>(options);
 
-    // we initialize the path with the link value
-    const [path, setPath] = useState<string | false | undefined>(() => {
```

Comment:
I wonder: Why did we add this callback in the first place since the effect below does the same thing? Some `git blame` may reveal a hidden reason.

Reply:
Done in https://github.com/marmelab/react-admin/commit/d57d112bcca00db4ea499537d663b4496c8f7982 when we added support for async link functions. I believe it should not have been merged as is